packages/next/src/build/swc/index.ts

```diff
@@ -839,12 +848,56 @@ function bindingToApi(binding: any, _wasm: boolean) {
       )
     }
 
-    async changed(): Promise<AsyncIterableIterator<TurbopackResult>> {
-      const iter = subscribe<TurbopackResult>(false, async (callback) =>
-        binding.endpointChangedSubscribe(await this._nativeEndpoint, callback)
-      )
-      await iter.next()
-      return iter
+    async changed(): Promise<
+      AsyncIterableIterator<TurbopackResult<ServerClientChange>>
+    > {
+      const nativeEndpoint = this._nativeEndpoint
+      return (async function* () {
+        const endpoint = await nativeEndpoint
+        let server: TurbopackResult | undefined
+        let client: TurbopackResult | undefined
+
+        const serverPromise = subscribe<TurbopackResult>(
+          false,
+          async (callback) =>
+            binding.endpointServerChangedSubscribe(endpoint, callback)
+        )
```

Comment:
Do we need to re-subscribe every time? That might introduce a memory leak if we don't properly dispose of the subscription.

Instead, we could keep both subscriptions alive and race `let nextClient = clientSubscription.next(); let nextServer = serverSubscription.next()`?

Reply:
Resubscribing also adds a race condition, since it subscribes to events starting from the time of the subscription

Reply:
That's a great catch about subscription leaks and reuse. I've adjusted to reuse the subscription, and properly cleanup when we stop iterating.